evm/vm/message.py

```diff
@@ -142,3 +143,73 @@ def storage_address(self, value):
     @property
     def is_create(self):
         return self.to == CREATE_CONTRACT_ADDRESS
+
+
+class ShardingMessage(Message):
+
+    def __init__(self,
+                 gas,
+                 gas_price,
+                 to,
+                 sender,
+                 value,
+                 data,
+                 code,
+                 origin=None,
+                 access_list=None,
+                 depth=0,
+                 create_address=None,
+                 code_address=None,
+                 should_transfer_value=True,
+                 is_static=False):
+        validate_uint256(gas, title="Message.gas")
+        self.gas = gas
+
+        validate_uint256(gas_price, title="Message.gas_price")
+        self.gas_price = gas_price
+
+        validate_canonical_address(to, title="Message.to")
+        self.to = to
+
+        if sender != ENTRY_POINT:
```

Comment:
This check shouldn't be required since the `ENTRY_POINT` is a valid canonical address (at least I think it is)

Reply:
yeah `ENTRY_POINT` is a valid canonical address so we only validate if `sender` is not `ENTRY_POINT`

Reply:
Silly me.  I saw `==` even though it's clearly `!=`